src/global-table.ts

```diff
@@ -1,22 +1,88 @@
-import { IResource, RemovalPolicy, Resource, Stack } from 'aws-cdk-lib';
+import { IResource, RemovalPolicy, Resource } from 'aws-cdk-lib';
 import * as ddb from 'aws-cdk-lib/aws-dynamodb';
+import * as iam from 'aws-cdk-lib/aws-iam';
 import { Construct } from 'constructs';
+import * as perms from './perms';
 
 export interface IGlobalTable extends IResource {
+  grant(grantee: iam.IGrantable, ...actions: string[]): iam.Grant;
+  grantReadData(grantee: iam.IGrantable): iam.Grant;
+  grantWriteData(grantee: iam.IGrantable): iam.Grant;
+  grantReadWriteData(grantee: iam.IGrantable): iam.Grant;
 }
 
 abstract class GlobalTableBase extends Resource implements IGlobalTable {
+  public abstract readonly tableArn: string;
+  protected readonly regionalArns = new Array<string>();
+  public abstract readonly tableName: string;
+  public grant(grantee: iam.IGrantable, ...actions: string[]): iam.Grant {
+    return iam.Grant.addToPrincipal({
+      grantee,
+      actions,
+      resourceArns: [
+        this.tableArn,
+      ],
+      scope: this,
+    });
+  }
+
+  public grantReadData(identity: iam.IGrantable): iam.Grant {
+    return iam.Grant.addToPrincipal({
+      grantee: identity,
+      actions: perms.READ_DATA_ACTIONS.concat(perms.DESCRIBE_TABLE), // ['s3:Get*']as many actions as you need
+      resourceArns: [this.tableArn],
+    });
+  }
+
+  public grantWriteData(grantee: iam.IGrantable): iam.Grant {
+    //const tableActions = perms.WRITE_DATA_ACTIONS.concat(perms.DESCRIBE_TABLE);
+    //const keyActions = perms.KEY_READ_ACTIONS.concat(perms.KEY_WRITE_ACTIONS);
+    //return this.combinedGrant(grantee, { keyActions, tableActions });
+    return iam.Grant.addToPrincipal({
+      grantee,
+      actions: perms.WRITE_DATA_ACTIONS.concat(perms.DESCRIBE_TABLE),
+      resourceArns: [this.tableArn],
+    });
+  }
+
+  public grantReadWriteData(grantee: iam.IGrantable): iam.Grant {
+    //const tableActions = perms.READ_DATA_ACTIONS.concat(perms.WRITE_DATA_ACTIONS).concat(perms.DESCRIBE_TABLE);
+    //const keyActions = perms.KEY_READ_ACTIONS.concat(perms.KEY_WRITE_ACTIONS);
+    //return this.combinedGrant(grantee, { keyActions, tableActions });
+    return iam.Grant.addToPrincipal({
+      grantee,
+      actions: perms.READ_DATA_ACTIONS.concat(perms.WRITE_DATA_ACTIONS).concat(perms.DESCRIBE_TABLE),
+      resourceArns: [this.tableArn],
+    });
+  }
+
 }
 
 export interface GlobalTableProps {
   readonly partitionKey: ddb.Attribute;
+  readonly replicas ?: Replica[];
+  readonly tableName ?: string;
```

Comment:
Since a customer is able to supply a physical name for this resource, you must pass it through to the Constructs super constructor. See the Example Construct Library for an example.